apps/examples/smartfs/fs_performance/fs_performance_main.c

```diff
@@ -0,0 +1,522 @@
+				fprintf(stderr, "ERROR: Failed to get Free Run Timer status: %d\n", errno);
+				ret = -1;
+				goto close_with_frt;
+			}
+
+			for (i = 1; i <= ITR_READ; i++) {
+				fd = open(g_file[0], O_RDOK);
+				if (fd < 0) {
+					printf("Unable to open file: %s to read, fd: %d\n", g_file[0], fd);
+					ret = ERROR;
+					goto close_with_frt;
+				}
+
+				do {
+				
+					ret = read(fd, g_read_buf, g_bsz);
+				} while (ret == g_bsz);
+
+				close(fd);
+			}
+
+			if (ioctl(frt_fd, TCIOC_GETSTATUS, (unsigned long)(uintptr_t)&after) < 0) {
+				fprintf(stderr, "ERROR: Failed to get Free Run Timer status: %d\n", errno);
+				ret = -1;
+				goto close_with_frt;
+			}
+
+			tot_time += (after.timeleft - before.timeleft);
+			printf("End File size: %d bytes, Buffer size: %d bytes, Total time: %lu us\n", file_size, g_bsz, tot_time);
+
+			ret = unlink(g_file[0]);
+			if (ret < 0) {
+				printf("Failed to unlink file %s, errno: %d\n", g_file[0], errno);
+				ret = ERROR;
+				goto close_with_frt;
+			}
+		}
+		break;
+
+	case TEST_FLAG_GC_EFFECT :
+		printf("\n****************************************************************************************\n");
+		printf("**********************************TEST GC EFFECT************************************");
+		printf("\n****************************************************************************************\n");
+		for (i = 1; i <= ITR_GC; i++) {
+			if (ioctl(frt_fd, TCIOC_GETSTATUS, (unsigned long)(uintptr_t)&before) < 0) {
+				fprintf(stderr, "ERROR: Failed to get Free Run Timer status: %d\n", errno);
+				ret = -1;
+				goto close_with_frt;
+			}
+
+			fd = open(g_file[0], O_WROK | O_CREAT);
+			if (fd < 0) {
+				printf("Unable to open file: %s, fd: %d\n", g_file[0], fd);
+				ret = ERROR;
+				goto close_with_frt;
+			}
+
+			ret = write(fd, g_rw_buf, g_fsz);
+			if (ret != (g_fsz)) {
+				 printf("Unable to write to file: %s, ret: %d\n", g_file[0], ret);
+				 ret = ERROR;
+				 goto close_with_fd;
+			}
+			close(fd);
+
+			ret = unlink(g_file[0]);
+			if (ret != OK) {
+				printf("Failed to unlink file: %s, errno: %d\n", g_file[0], errno);
+				ret = ERROR;
+				goto close_with_frt;
+			}
+
+			if (ioctl(frt_fd, TCIOC_GETSTATUS, (unsigned long)(uintptr_t)&after) < 0) {
+				fprintf(stderr, "ERROR: Failed to get Free Run Timer status: %d\n", errno);
+				ret = -1;
+				goto close_with_frt;
+			}
+
+			itr_time = after.timeleft - before.timeleft;
+			printf("iteration time: %lu us\n", i, itr_time);
+			tot_time += itr_time;
+		}
+		printf("\nFinal time: %lu us\n", tot_time);
+		break;
+
+	default:
+		break;
+	}
+
+	if (ioctl(frt_fd, TCIOC_STOP, 0) < 0) {
+		fprintf(stderr, "ERROR: Failed to get Free Run Timer status: %d\n", errno);
+		ret = -1;
+		close(frt_fd);
+	}
+
+	close(frt_fd);
+	return ret;
+
+close_with_fd:
+	close(fd);
+close_with_frt:
+	if (ioctl(frt_fd, TCIOC_STOP, 0) < 0) {
+		fprintf(stderr, "ERROR: Failed to stop the Free Run Timer: %d\n", errno);
+	}
+
+	if (frt_fd > 0) {
```

Comment:
This check should be done before `ioctl(frt_fd. TCIOC_STOP, 0)`
Or rather, if frt_fd is not able top be opened, code flow should skip this automatically.

Reply:
This check is not required since it frt_fd is not valid it will exit in the first step itself.
I will remove it.